Assets/_BForBoss/_Core/Scripts/Managers/WeaponRaceCourseWorldManager.cs

```diff
@@ -0,0 +1,122 @@
+using System.Linq;
+using Perigon.Analytics;
+using Perigon.Entities;
+using Perigon.Leaderboard;
+using Perigon.Utility;
+using Perigon.Weapons;
+using Sirenix.OdinInspector;
+using Sirenix.Utilities;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+namespace BForBoss
+{
+    public class WeaponRaceCourseWorldManager: BaseWorldManager
+    {
+        [Title("Component")] 
+        [SerializeField] private TimeManager _timeManager = null;
+        [SerializeField] private CheckpointManager _checkpointManager = null;
+        [Title("Weapon/Equipment Component")] 
+        [SerializeField] private WeaponsManager _weaponsManager = null;
+        [SerializeField] private EquipmentBehaviour _equipmentBehaviour = null;
+        [SerializeField] private AmmunitionCountViewBehaviour _ammunitionCountView = null;
+        [SerializeField] private ReloadViewBehaviour _reloadView = null;
+        
+        [Title("User Interface")]
+        [SerializeField] private TimerViewBehaviour _timerView = null;
+        [SerializeField] private ForcedSetUsernameViewBehaviour _forcedUploadView = null;
+
+        [Title("Effects")] 
+        [SerializeField] private Volume _deathVolume = null;
+
+        private readonly PerigonAnalytics _perigonAnalytics = PerigonAnalytics.Instance;
+        private PostProcessingVolumeWeightTool _postProcessingVolumeWeightTool = null;
+        private DetectInput _detectInput = new DetectInput(); //Placeholder, remove this after finishing the timed leader board stuff
+        private readonly TimeManagerViewModel _timeManagerViewModel = new TimeManagerViewModel();
+        private UploadPlayerScoreDataSource _uploadPlayerScoreDataSource = null;
+        private LifeCycleBehaviour[] _lifeCycleBehaviours = null;
+        
+        protected override Vector3 SpawnLocation => _checkpointManager.CheckpointPosition;
+        protected override Quaternion SpawnLookDirection => _checkpointManager.CheckpointRotation;
+
+        protected override void Reset()
+        {
+            _checkpointManager.Reset();
+            base.Reset();
+            _detectInput.Reset();
+            _timeManager.Reset();
+            _timerView.Reset();
+            _lifeCycleBehaviours.ForEach(dummy => dummy.Reset());
+        }
+
+        protected override void Awake()
+        {
+            base.Awake();
+            _postProcessingVolumeWeightTool = new PostProcessingVolumeWeightTool(_deathVolume, 0.1f, 0f, 0.1f);
```

Comment:
I may have asked this before, but should we cache the vector3 as a readonly variable

Reply:
Oh these aren't Vector3, they are
duration, startValue, endValue

Reply:
oh yeah, whoops brain fart. then `0.1f` should be a const or something, rather than a magic number